### PR TITLE
22 apis product module

### PR DESCRIPTION
**Pull request by SruthiKonda12**
Please review the code

### CODE REVIEW

File: apps/masters/models.py [hunk 1]
```diff
@@ -1,8 +1,63 @@
...
+    class Meta:
+        db_table = countrytable
+
+class State(models.Model):
+    state_id = models.AutoField(primary_key=True)
+    country_id = models.ForeignKey(Country, on_delete=models.CASCADE, null=True, default=None, db_column = 'country_id')
+    state_name = models.CharField(max_length=100)
+    state_code = models.CharField(max_length=10)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```
Comment (thulasiram1216) [inline apps/masters/models.py:27]: This should satisfy below statement, did you test?
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP

File: apps/masters/serializers.py [hunk 2]
```diff
@@ -75,45 +116,74 @@ class Meta:
         model = Transporters
         fields = ['transporter_id', 'name', 'code', 'gst_no', 'website_url']
```
Comment (thulasiram1216) [inline apps/masters/serializers.py:117]: what are these fields? why only few fields inthe list?

File: apps/masters/serializers.py [hunk 3]
```diff
@@ -75,45 +116,74 @@ class Meta:
...
         fields = '__all__'
 
-class CitySerializer(serializers.ModelSerializer):
-    state = ModStateSerializer(source='state_id', read_only = True)
+
+class ModBrandSalesmanSerializer(serializers.ModelSerializer):
     class Meta:
-        model = City
+        model = BrandSalesman
+        fields = ['brand_salesman_id','code','name','commission_rate','rate_on']
```
Comment (thulasiram1216) [inline apps/masters/serializers.py:148]: what are these fields? why only few fields inthe list?

File: apps/masters/serializers.py [hunk 4]
```diff
@@ -75,45 +116,74 @@ class Meta:
...
+    class Meta:
+        model = BrandSalesman
         fields = '__all__'
+
+
+
+class ModProductBrandsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ProductBrands
+        fields = ['brand_id','brand_name','code']
```
Comment (thulasiram1216) [inline apps/masters/serializers.py:160]: what are these fields? why only few fields inthe list?

File: apps/products/models.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+from django.db import models
+from apps.masters.models import *
+from django.dispatch import receiver
+from django.db.models.signals import pre_delete
+from utils_methods import EncryptedTextField
+from utils_variables import *
+
+
+def product_groups_picture(instance, filename):
+    # Get the file extension
```
Comment (thulasiram1216) [inline apps/products/models.py:12]: why this method is outside the class? if it is common method and can be reuesed then move to utils_methods.py file

File: apps/products/models.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+    is_loan_account = models.BooleanField(default=False)
+    tds_applicable = models.BooleanField(default=False)
+    address = models.CharField(max_length=255)
+    pan = models.CharField(max_length=50)
+    employee = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = 'product_sales_gl'
```
Comment (thulasiram1216) [inline apps/products/models.py:145]: db_table value should come from utils_variables.py

File: apps/products/models.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+    def __str__(self):
+        return f"{self.sales_gl_id} {self.name}"
+
+class ProductPurchaseGl(models.Model):
```
Comment (thulasiram1216) [inline apps/products/models.py:150]: Why this class is duplicated?

File: apps/products/models.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = 'product_purchase_gl'
+
+    def __str__(self):
+        return f"{self.purchase_gl_id} {self.name}"
+
+
+def products_picture(instance, filename):
```
Comment (thulasiram1216) [inline apps/products/models.py:177]: why this method is outside the class? if it is common method and can be reuesed then move to utils_methods.py file

File: apps/products/serializers.py [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+    ledger_group = LedgerGroupsSerializer(source='ledger_group_id',read_only=True)
+    class Meta:
+        model = ProductPurchaseGl
+        fields = '__all__'
+	
+
+class ModproductsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = products
+        fields = ['product_id','name']
```
Comment (thulasiram1216) [inline apps/products/serializers.py:121]: why only 2 fields here?

File: apps/products/urls.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+router = routers.DefaultRouter()
+router.register(r'product_groups', ProductGroupsViewSet)
+router.register(r'product_categories', ProductCategoriesViewSet)
+router.register(r'product_stock_units', ProductStockUnitsViewSet)
+router.register(r'product_gst_classifications', ProductGstClassificationsViewSet)
+router.register(r'product_sales_gl', ProductSalesGlViewSet)
+router.register(r'product_purchase_gl', ProductPurchaseGlViewSet)
+router.register(r'products', productsViewSet)
+
```
Comment (thulasiram1216) [inline apps/products/urls.py:14]: follow the pattern where pavan and pramod follwed

File: apps/masters/models.py (hunk 1 above)
Comment (SruthiKonda12) [reply apps/masters/models.py:27]: yes tested Thulasi  it is satisfied.

File: apps/masters/serializers.py (hunk 2 above)
Comment (SruthiKonda12) [reply apps/masters/serializers.py:117]: It is done by pramod

File: apps/products/models.py (hunk 5 above)
Comment (SruthiKonda12) [reply apps/products/models.py:12]: Actually from picture column we are calling that method Thulasi
    picture = models.ImageField(max_length=255, default=None, null=True, upload_to=product_groups_picture)
and we tried putting it in utils_methods but everyone having different requirements so we left like that only .

File: apps/products/models.py (hunk 6 above)
Comment (SruthiKonda12) [reply apps/products/models.py:145]: Changed Thulasi

File: apps/products/models.py (hunk 7 above)
Comment (SruthiKonda12) [reply apps/products/models.py:150]: No duplicate class is created Thulasi 
ProductSalesGl
ProductPurchaseGl
both are different classes Thulasi

File: apps/products/models.py (hunk 8 above)
Comment (SruthiKonda12) [reply apps/products/models.py:177]: Actually from picture column we are calling that method Thulasi
    picture = models.ImageField(max_length=255, default=None, null=True, upload_to=products_picture)
and we tried putting it in utils_methods but everyone having different requirements so we left like that only .

File: apps/products/urls.py (hunk 10 above)
Comment (SruthiKonda12) [reply apps/products/urls.py:14]: Changed Thulasi

File: apps/products/serializers.py (hunk 9 above)
Comment (SruthiKonda12) [reply apps/products/serializers.py:121]: This serializer is a duplicate of product serializer, if anyone getting FK  from in this table  they will get only 2 fileds in data format.

File: apps/masters/serializers.py (hunk 4 above)
Comment (SruthiKonda12) [reply apps/masters/serializers.py:160]: Removed duplicate serializer Thulasi

File: apps/masters/serializers.py (hunk 3 above)
Comment (SruthiKonda12) [reply apps/masters/serializers.py:148]: Removed Thulasi

**Pull request merged**